.circleci/config.yml

```diff
@@ -132,16 +97,14 @@ jobs:
 
   publish:
     docker:
-      - image: udata/circleci:2-alpine
+      - image: cimg/python:3.7.13
     steps:
       - attach_workspace:
           at: .
       - deploy:
           name: Publish on PyPI
           command: |
-            source venv/bin/activate
-            pip install twine
-            twine upload --username "${PYPI_USERNAME}" --password "${PYPI_PASSWORD}" dist/*.whl
+            poetry publish --username "${PYPI_USERNAME}" --password "${PYPI_PASSWORD}" --no-interaction
```

Comment:
Could you test an upload in CI by adding this branch as a filter for `publish`? You could try to push on https://test.pypi.org/.